### PR TITLE
Expose --delay and --max-attempts for wait commands (#1295)

### LINKED ISSUE #1295: Expose parameters for 'delay' and 'max-attempts'
I'm trying to migrate to the unified cli, and to use waiters instead of polling loops, but the conversion-task-completed waiter consistently times out in us-east-1 for my 12GB disk.

It reports 
Waiter ConversionTaskCompleted failed: Max attempts exceeded
but the task does complete successfully. I can write a loop which calls the waiter, but I'd prefer to increase the number of attempts, or reduce their frequency. They appear to be taken from botocore/data/aws/ec2/2014-09-01.waiters.json, which contains the following:

"ConversionTaskCompleted": {      "delay": 15,      "operation":
"DescribeConversionTasks",      "maxAttempts": 40,

I'd like these to be exposed as parameters in the aws cli command, or perhaps for the timeout to be derived from the size of the disk being converted.

### PR DESCRIPTION
**Pull request by minjcho**
*Issue #, if available:*
Fixes #1295 

*Description of changes:*
* Add ``--delay`` and ``--max-attempts`` arguments to waiter commands.
* Pass the provided values through ``WaiterConfig`` when invoking ``waiter.wait()``.
* Update waiter help text to document the new options.
* Add unit coverage for argument parsing, help output, and waiter config injection.

*Description of tests:*
* Added unit tests covering argument parsing, help output, and waiter config injection. 
* Ran `python -m pytest tests/unit/customizations/test_waiters.py -q` — all pass.
* Succeeded manual testing described below. 
 
*Manual testing:*

Verified the new flags appear in the generated help output:

```
aws ec2 wait instance-running help
```

I confirmed that `[--delay <value>]` and `[--max-attempts <value>]` are listed in the synopsis and documented in the options section.

Then I ran:

```
aws ec2 wait instance-running --instance-ids <YOUR_INSTANCE_ID> --delay 5 --max-attempts 10
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by minjcho** [comment]
**Added input validation for `--delay` and `--max-attempts`**

Since other CLI params don't check non-positive values at the CLI layer (they rely on botocore's `ParamValidator` or the AWS server to reject them), I initially didn't add validation here either.

But I found that `WaiterConfig` is a special case — it's popped from kwargs via `kwargs.pop('WaiterConfig', {})` before reaching `ParamValidator` or the API server, and consumed locally in `time.sleep()` and a loop counter. **No layer validates these values:**

- `--delay -1` → `ValueError: sleep length must be non-negative` (Python runtime error from `time.sleep(-1)`)
- `--delay 0` → rapid-fire API calls with no sleep interval
- `--max-attempts 0` or `-1` → polls once, then immediately raises `WaiterError`

Added validation in `WaiterArgument.add_to_params` matching the waiter JSON schema constraints (`delay >= 0`, `maxAttempts >= 1` from `botocore/tests/functional/test_waiter_config.py`):

```bash
$ aws ec2 wait instance-running --instance-ids i-123 --delay -1
aws: [ERROR]: --delay must be a non-negative integer, got -1

$ aws ec2 wait instance-running --instance-ids i-123 --max-attempts 0
aws: [ERROR]: --max-attempts must be a positive integer, got 0
```

Added 7 unit tests covering rejection and acceptance boundaries.

I'll also file a separate issue on botocore for adding the same validation in `Waiter.wait()`, since boto3 users are equally affected.

File: .changes/next-release/waiter-delay-max-attempts.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+[
+  {
+    "category": "waiter",
+    "description": "Add ``--delay`` and ``--max-attempts`` arguments to all ``wait`` commands, allowing users to override the default polling interval and maximum number of attempts.",
```
Comment (aemous) [inline .changes/next-release/waiter-delay-max-attempts.json:4]: This changelog entry looks hand-made. Did you use our automated script?

Please run `./scripts/new-change` to automatically generate a changelog entry with the proper format and name. Then, add the generated changelog entry to this PR and delete this hand-made one.

File: tests/unit/customizations/test_waiters.py [hunk 2]
```diff
@@ -166,6 +167,13 @@ def test_wait_state_help_command(self):
         self.assert_contains('``--filters`` (list)')
         self.assert_contains('======\nOutput\n======\n\nNone')
 
+    def test_wait_state_help_shows_waiter_params(self):
+        self.driver.main(['ec2', 'wait', 'instance-running', 'help'])
+        self.assert_contains('[--delay <value>]')
+        self.assert_contains('[--max-attempts <value>]')
+        self.assert_contains('``--delay``')
+        self.assert_contains('``--max-attempts``')
```
Comment (aemous) [inline tests/unit/customizations/test_waiters.py:175]: These two assertions are redundant, right?

If the output contains `[--delay <value>]`, the since `[--delay <value>]` contains `--delay`, we immediately know the output contains `--delay`.

Please either remove the redundant assertions or update them so they are not redundant.

File: tests/unit/customizations/test_waiters.py [hunk 3]
```diff
@@ -203,6 +211,53 @@ def test_rds_jobs_complete(self):
...
+        cmdline += ' --delay 10 --max-attempts 100'
+        # WaiterConfig is popped by botocore's Waiter.wait() before
+        # the API call, so only API params appear in last_kwargs.
+        result = {'InstanceIds': ['i-12345678']}
+        self.parsed_response = {
+            'Reservations': [{'Instances': [{'State': {'Name': 'running'}}]}]
+        }
+        self.assert_params_for_cmd(cmdline, result)
+
+    def test_ec2_wait_forwards_waiter_config_end_to_end(self):
```
Comment (aemous) [inline tests/unit/customizations/test_waiters.py:226]: This is not a unit test since it is not testing a single class/function in isolation. Please either move this test under `tests/functional/test_waiters.py` or update the test so that it is a unit test.

The same is true for `test_ec2_wait_with_delay_and_max_attempts` and `test_ec2_wait_without_waiter_flags_omits_waiter_config`.

File: awscli/customizations/waiters.py [hunk 4]
```diff
@@ -203,7 +272,9 @@ def _build_polling_description(self, delay, max_attempts):
         description = (
             ' It will poll every %s seconds until a successful state '
             'has been reached. This will exit with a return code of 255 '
-            'after %s failed checks.' % (delay, max_attempts)
+            'after %s failed checks. You can override the default polling '
+            'behavior with ``--delay`` and ``--max-attempts``.'
+            % (delay, max_attempts)
```
Comment (aemous) [inline awscli/customizations/waiters.py:277]: Can we switch from using the formatting operator `%` to f-strings to better align with our formatting?

File: tests/unit/customizations/test_waiters.py [hunk 5]
```diff
@@ -399,3 +470,129 @@ def test_invoke(self):
...
+        waiter_caller.invoke(
+            'myservice', 'MyWaiter', parameters, parsed_globals
+        )
+
+        waiter.wait.assert_called_with(
+            Foo='bar',
+            WaiterConfig={'Delay': 10, 'MaxAttempts': 50},
+        )
+
+    def test_invoke_without_waiter_config(self):
```
Comment (aemous) [inline tests/unit/customizations/test_waiters.py:497]: Isn't this test redundant to the existing `test_invoke`? `test_invoke` does not supply waiter params, and it uses `assert_called_with`, so if any waiter params are supplied to `wait`, it would fail with an `AssertionError`.

File: tests/unit/customizations/test_waiters.py [hunk 6]
```diff
@@ -399,3 +470,129 @@ def test_invoke(self):
...
+        waiter_caller = WaiterCaller(session, waiter_name)
+        waiter_caller.invoke(
+            'myservice', 'MyWaiter', parameters, parsed_globals
+        )
+
+        # Without set_waiter_config, no WaiterConfig should be passed.
+        waiter.wait.assert_called_with(Foo='bar')
+
+
+class TestWaiterArgument(unittest.TestCase):
```
Comment (aemous) [inline tests/unit/customizations/test_waiters.py:518]: For net-new test suites let's use PyTest instead of extending the older `unittest.TestCase` base class.

i.e. migrate `TestWaiterArgument` to use PyTest.

File: tests/unit/customizations/test_waiters.py [hunk 7]
```diff
@@ -399,3 +470,129 @@ def test_invoke(self):
...
+        self.assertEqual(arg.cli_type_name, 'integer')
+
+    def test_required_defaults_to_false(self):
+        arg = WaiterArgument('delay', 'help text', 'Delay')
+        self.assertFalse(arg.required)
+
+    def test_delay_rejects_negative(self):
+        arg = WaiterArgument('delay', 'help text', 'Delay')
+        params = {}
+        with self.assertRaises(Exception) as ctx:
```
Comment (aemous) [inline tests/unit/customizations/test_waiters.py:563]: Can we narrow down the scope of the expected Exception to the actual exception that gets raised (`ParamValidationError`)?

Similar comment for `test_delay_rejects_negative_large`, `test_max_attempts_rejects_zero`, and `test_max_attempts_rejects_negative`.

File: tests/unit/customizations/test_waiters.py [hunk 8]
```diff
@@ -166,6 +167,13 @@ def test_wait_state_help_command(self):
         self.assert_contains('``--filters`` (list)')
         self.assert_contains('======\nOutput\n======\n\nNone')
```
Comment (aemous) [inline tests/unit/customizations/test_waiters.py:169]: I saw we added tests that validate there's an error when passing negative values for delay/max-attempts to `WaiterCaller`. Can we also add functional tests that validate there's an error end-to-end when running a command that uses negative values for these new params? The new test(s) would live in `tests/functional/test_waiters.py`.

**Review by aemous — CHANGES_REQUESTED** [review]
Thanks for the contribution. I requested some changes and asked a few questions.

**Comment by RyanFitzSimmonsAK** [comment]
I'm setting this pull request to Draft due to author inactivity.  If you implement the requested changes, please feel free to reopen it.
